lib/rules/declaration-block-no-redundant-longhand-properties/index.js

```diff
@@ -28,6 +28,43 @@ const meta = {
 
 /** @type {Map<string, (decls: Map<string, Declaration>) => (string | undefined)>} */
 const customResolvers = new Map([
+	[
+		'font-synthesis',
+		(decls) => {
+			const weight = decls.get('font-synthesis-weight')?.value.trim();
+			const style = decls.get('font-synthesis-style')?.value.trim();
+			const smallCaps = decls.get('font-synthesis-small-caps')?.value.trim();
+
+			/** @type {(s: string | undefined) => boolean} */
+			const isValidFontSynthesisValue = (s) => s === 'none' || s === 'auto';
+
+			if (
+				!isValidFontSynthesisValue(weight) ||
+				!isValidFontSynthesisValue(style) ||
+				!isValidFontSynthesisValue(smallCaps)
+			) {
+				return;
+			}
```

Comment:
I realized that earlier, this line was uncovered. I've added a test that involves some invalid CSS (in https://github.com/stylelint/stylelint/pull/7214/commits/8fddced20cb68a0b539cce463d1f319abb49ad48) - is this the right way to go about this? 

(I noticed some of the earlier test cases also involve slightly malformed CSS, ex the transition test cases)

If not, I can remove the test case and/or try something else?

Reply:
> which are the only allowed values in the spec

I think that global values are also accepted.
i.e. inherit; initial; revert; revert-layer; unset;

I wonder if `decls.get` retrieves it raw or if it does the mapping automatically.
i.e. is `font-synthesis-weight: initial;` considered invalid? (should map to `auto` and hence be considered valid)

Reply:
I think the added test case is enough to cover the path. 👍🏼 

And supporting global values is ideal, but I don't think it's required for now. We can revisit it later if people request.

Reply:
> We can revisit it later if people request.

It should be done for all properties at once; let's leave it for a later enhancement.